### ISSUE TITLE
Termux-exec allows execution of programs lacking permissions

### ISSUE THREAD

**Issue by Neo-Oli** [termux/termux-exec#5]
Termux-exec ignores missing permissions of a file. It allows files that don't have executive permissions to be executed anyway. This only happens when the file has a shebang that requires termux-exec. Normal termux-shebangs still give the correct 'permission denied'.

```
$ cat test.sh
#!/bin/sh
echo Hello World
$ ./test.sh
Hello World
$ chmod -x test.sh
$ ./test.sh
Hello World
$ ls -l test.sh
-rw------- 1 u0_a401 u0_a401 27 Sep 28 10:48 test.sh
$ termux-fix-shebang test.sh
$ ./test.sh
zsh: permission denied: ./test.s
```

**Comment by fornwall** [comment]
Good point! What do you think about https://github.com/termux/termux-exec/pull/6 ?

**Comment by Neo-Oli** [comment]
That seems to work. I was afraid at first that it would still be able to execute things on the SD Card, because the permissions seem to imply executable, but it still correctly fails.

On a side note: This bug enabled me to execute scripts on my SD Card, which seems to be what a lot of people want. So maybe this could be implemented* as a feature in a different program. It's probably out of scope for this one.

*Making it work on both  normal and Termux shebangs.

**Comment by fornwall** [comment]
Should be fixed now as of version 0.3 of the termux-exec package.
